client/components/Calendar.jsx

```diff
@@ -14,26 +14,57 @@ const Calendar = (props) => {
     [0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0],
   ];
-  // get data for first day of current month
-  const first = new Date(props.year, props.month, 1);
+
+  const monthFirstDay = new Date(props.year, props.month, 1);
+  const monthLastDay = new Date(props.year, props.month + 1, 0);
+  const currentMonth = new Date().getMonth();
+
+  const bookingsMonthIndex = props.month < currentMonth ? currentMonth + props.month : props.month - currentMonth;
+  let bookedDates = props.bookedDates[bookingsMonthIndex].slice().sort((a, b) => b - a);
+  //declare counter to fill table with dates
+  let dateCounter = 1;
   
-  //get data for last day of current month
-  const last = new Date(props.year, props.month + 1, 0);
+  const renderDateCell = function (currentCell) {
+    if (dateCounter > monthLastDay.getDate()) {
+      return null;
+    }
+    const currentCellDate = new Date(props.year, props.month, dateCounter);
+    let classNames = '';
+    let clickHandler = (e) => props.selectDate(+e.target.innerHTML);
+    if (currentCell === 0) {
+      classNames += 'day ';
+    } 
+    if ((props.requestedDates[0] !== undefined && props.requestedDates[0].toDateString() === currentCellDate.toDateString()) || (props.requestedDates[1] !== undefined && props.requestedDates[1].toDateString() === currentCellDate.toDateString())) {
```

Comment:
Maybe consider breaking all of the class-name conditionals into a separate function, e.g., getClassNamesForCell? This looks a lot cleaner than yesterday, but I think that extra step would help communicate even more clearly.

Reply:
Totally agree with this. I considered doing it now but I may move on and come back to the calendar. Thanks again for your feedback.